pkg/controllers/restore/controller.go

```diff
@@ -356,10 +356,15 @@ func (h *handler) generateDependencyGraph(ownerToDependentsList map[string][]res
 		resourceInfoToData = objFromBackupCR.namespacedResourceInfoToData
 	}
 	for resourceInfo, resourceData := range resourceInfoToData {
-		// add to adjacency list
 		name := resourceInfo.Name
 		namespace := resourceInfo.Namespace
 		gvr := resourceInfo.GVR
+		if resourceData.GetKind() == "Deployment" && namespace == "cattle-system" {
+			if strings.HasSuffix(name, "rancher") || strings.HasSuffix(name, "rancher-webhook") {
```

Comment:
Why use a substring comparison over a direct one?

Reply:
If the user sets a different name when installing rancher, for example, `helm install my-setup rancher-latest/rancher`, the deployment’s name will be `my-setup-rancher`. `strings.HasSuffix(name, "rancher")` is to catch this case. 
So does `strings.HasSuffix(name, "rancher-webhook")`